src/chkjson/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+
+set(CHKJSON_SOURCES
+	${CMAKE_SOURCE_DIR}/src/json.cpp
+	${CMAKE_SOURCE_DIR}/src/chkjson/chkjson.cpp
+)
+
+SET (CHKJSON_HEADERS
+	${CMAKE_SOURCE_DIR}/src/json.h
+)
+
+INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/src/chkjson )
+
+# Add the actual executable
+ADD_EXECUTABLE(chkjson ${CHKJSON_SOURCES} ${CHKJSON_HEADERS})
```

Comment:
This adds the chkjson file to the "all" target and it's therefor made every time somebody runs `make all`. It's not a big thing (only 2 source files + linking), but the original Makefile has this as a separate target that has to be invoked explicitly.